tatorte_classifier/api/texts.py

```diff
@@ -0,0 +1,139 @@
+import logging
+
+from flask import jsonify, request
+from flask_restful import Resource
+from bson.json_util import dumps
+
+from werkzeug.exceptions import BadRequest
+
+from tatorte_classifier.database import (
+    get_all_texts,
+    get_text,
+    modify_text,
+    create_text,
+    get_random_text,
+    delete_text,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class Text(Resource):
+    def get(self, text_id):
+        """
+        Input:
+            text_id: [id of text, 'random']
+        Returns:
+            {
+                "_id": {
+                    "$oid": "5c6c1b2573cda500b254404c"
+                },
+                "data": "This is a test. Number 2",
+                "time_created": "2019-02-19 15:05:09",
+                "time_modified": "2019-02-19 15:18:53",
+                "categories": [4, 2]
+            }
+        """
+        try:
+            if text_id == "random":
+                return dumps(get_random_text())
+            return dumps(get_text(text_id))
+        except Exception as err:
+            logger.error(str(err))
+            return BadRequest(str(err))
+
+    def delete(self, text_id):
+        """Deletes text document with id == <text_id>
+        """
+
+        try:
+            delete_text(text_id)
+            return jsonify({"success": True})
+        except Exception as err:
+            logger.error(str(err))
+            return BadRequest(str(err))
+
+    def patch(self, text_id):
+        """
+        Input:
+            {
+                "categories": [3, 2],
+            }
+
+        Returns:
+            <id>
+        """
+        try:
+            request_json = request.get_json()
+            modify_text(text_id, request_json["categories"])
+            return jsonify({"success": True})
+        except Exception as err:
+            logger.error(str(err))
+            return BadRequest(str(err))
+
+
+class Texts(Resource):
```

Comment:
Same as with the `Models` resource: renaming to `TextList` makes it more readable and makes so output structure more predictable (surprise surprise: a list).